src/datatypes/datetime.ts

```diff
@@ -354,20 +408,20 @@ class DateTime {
         ? this.timezoneOffset * 60
         : new jsDate().getTimezoneOffset() * -1;
     return luxon.DateTime.fromObject({
-      year: this.year,
-      month: this.month,
-      day: this.day,
-      hour: this.hour,
-      minute: this.minute,
-      second: this.second,
-      millisecond: this.millisecond,
+      year: this.year ?? undefined,
+      month: this.month ?? undefined,
+      day: this.day ?? undefined,
+      hour: this.hour ?? undefined,
+      minute: this.minute ?? undefined,
+      second: this.second ?? undefined,
+      millisecond: this.millisecond ?? undefined,
```

Comment:
Short answer: yes, they are there because Luxon's type declarations say that the arguments of `fromObject` are `number | undefined`, and in our `DateTime` class, they are number or null.

The `??` is there to ensure that `0` is not treated as a falsy value (which broke some tests).

Is there a way to more nicely ensure that the luxon types are happy with what we have defined in each of these properties for datetime?

Reply:
I think what you did is right.  No changes needed.